src/main/java/com/atlauncher/gui/components/ImagePanel.java

```diff
@@ -22,31 +22,62 @@
 import java.awt.Graphics;
 import java.awt.Graphics2D;
 import java.awt.Image;
+import java.util.concurrent.Callable;
 
+import javax.annotation.Nonnull;
+import javax.swing.JLabel;
 import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+
+import com.atlauncher.managers.LogManager;
+import com.atlauncher.utils.Utils;
 
-@SuppressWarnings("serial")
 public final class ImagePanel extends JPanel {
     private static final Cursor HAND = new Cursor(Cursor.HAND_CURSOR);
+    private static final int DEFAULT_WIDTH = 300, DEFAULT_HEIGHT = 150;
 
     private volatile Image image;
 
-    public ImagePanel(Image image) {
-        this.image = image;
+    /**
+     * @param imageToLoad Deferred image loading
+     */
+    public ImagePanel(@Nonnull Callable<Image> imageToLoad) {
         this.setCursor(HAND);
-        this.setPreferredSize(new Dimension(Math.min(image.getWidth(null), 300), Math.min(image.getWidth(null), 150)));
+
+        add(new JLabel(Utils.getIconImage("/assets/image/loading-bars.gif")));
```

Comment:
I wouldn't show loading bars here, there's no network call, and loading from disk should be quick enough, the loading bars will look out of place